homeassistant/components/device_tracker/o2box.py

```diff
@@ -0,0 +1,226 @@
+"""
+Module to communicate with a O2 Box 1421, similar to https://hilfe.o2online.de/docs/DOC-1332
+"""
+
+import logging
+from datetime import timedelta
+from collections import namedtuple
+import requests
+
+from homeassistant.const import CONF_HOST, CONF_PASSWORD
+from homeassistant.helpers import validate_config
+from homeassistant.util import Throttle
+from homeassistant.components.device_tracker import DOMAIN
+
+MIN_TIME_BETWEEN_SCANS = timedelta(seconds=5)
+
+wlandevice = namedtuple('WlanDevice', ['mac', 'name', 'ip', 'signal', 'link_rate'])
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def get_scanner(hass, config):
+    """ Validates config and returns O2BoxScanner. """
+    if not validate_config(config,
+                           {DOMAIN: []},
+                           _LOGGER):
+        return None
+
+    scanner = O2BoxScanner(config[DOMAIN])
+    return scanner if scanner.success_init else None
+
+
+class O2BoxScanner(object):
+    """This class queries the O2Box"""
+    def __init__(self, config):
+        self.last_results = []
+        self.password = ''
+        self.success_init = True
+
+        # Check for user specific configuration
+        if CONF_HOST in config.keys():
```

Comment:
Also, why store host and password at the class level here? They are not used outside the init method.